util/src/test/java/io/kubernetes/client/PortForwardTest.java

```diff
@@ -121,13 +128,13 @@ public void run() {
                   }
                 } catch (IOException ex) {
                   ex.printStackTrace();
-                }
-              }
+                }}
             });
     synchronized (block) {
       t.start();
       block.wait();
     }
+    s.release();
```

Comment:
the release should happen inside the Thread. The idea is that you reserve the Semaphore before the thread starts, and then release it inside the thread. That way you know for sure that the Thread has executed before moving on to the test.

That can replace the sleep which was trying to do the same thing.